Chargebee/Classes/Cache/CBCache.swift

```diff
@@ -0,0 +1,87 @@
+//
+//  CBCache.swift
+//  Chargebee
+//
+//Created by ramesh_g on 28/07/23.
+//
+
+import Foundation
+
+
+struct ConfigCacheModel: Codable {
+    var config: CBAuthentication
+    var date: Date
+    init(config: CBAuthentication, date: Date) {
+        self.config = config
+        self.date = date
+    }
+}
+
+protocol CacheProtocal {
+    func writeConfigDetails(key:String,object:CBAuthentication)
+    func readConfigDetails(key:String,logger: CBLogger, handler: @escaping CBAuthenticationHandler)
+    func saveAuthenticationDetails(key:String,data: CBAuthenticationStatus)
+    func isCacheDataAvailable(key:String)-> Bool
+}
+
+struct CBCache: CacheProtocal {
+    
+    static let shared = CBCache()
+    var todaysDate = NSDate()
```

Comment:
Where is this being reset?

Reply:
> Where is this being reset?

`createTime()` This will create  24 Hours  time  and this date and time will be saved to `UserDefaults` 
`ConfigCacheModel(config: configObject, date: createTime())`

Comparison happens at this line  current time with the Saved time 
 ```
 if (self.todaysDate.compare(waitingDate as Date) == ComparisonResult.orderedDescending) {
//clearCacheFromMemory
}
```
Once the API call hits then time is created and it clears / Resets once time is completed.

Reply:
I get the logic, but where do we reset the `todaysDate` Since the `todaysDate` is an instance variable and is created when the static CBCache object is created.

Reply:
`todaysDate ` removed this property and added Date() directly update the code thanks